src/main/java/filemanager/directorytracker/ScheduleFileDirectory.java

```diff
@@ -70,6 +77,7 @@ public void goThroughToCheckFile(Command command) {
                     } catch (IOException e) {
                         e.printStackTrace();
                     }
+                    jobWriterService.saveJobInformation(clientName, command);
```

Comment:
Please introduce Command Type and differentiate user initiated commands from automatic (by setting different jobTypes for Command before passing it to this method). Also there is no need to write information about job execution for every client. That might be great to store info about client exports, but in a separate collection. I don't mind if you create one, like ExportStats and would write there owning Job ID, client name and number of exported transactions.

In general I think that information about Jobs shouldn't be written inside of this service. Job should contain Status field and be stored in database with status RUNNING as soon as they get picked up, and later updated to COMPLETE when related job is done (you need a separate write for this where you should add some additional information, at least the time when the job has been completed). Also you have to be sure that exceptions and stuff like that don't leave your database in inconsistent state. So please catch those and update status for problematic jobs to FAILED. If you could also catch cases when the app is shutting down and mark jobs as INTERRUPTED before exitting.

All that logic may be extracted to some Service that'd use FileDirectory.